lib/litmus/type/date_time.ex

```diff
@@ -0,0 +1,71 @@
+defmodule Litmus.Type.DateTime do
+  @moduledoc """
+  This type validates and converts ISO-8601 datetime with timezone strings into
+  `DateTime`s.
+
+  ## Options
+
+    * `:required` - Setting `:required` to `true` will cause a validation error
+      when a field is not present or the value is `nil`. Allowed values for
+      required are `true` and `false`. The default is `false`.
+
+  ## Examples
+
+      iex> schema = %{"start_date" => %Litmus.Type.DateTime{}}
+      iex> {:ok, %{"start_date" => datetime}} = Litmus.validate(%{"start_date" => "2017-06-18T05:45:33Z"}, schema)
+      iex> datetime
+      #DateTime<2017-06-18 05:45:33Z>
+
+  """
+
+  alias Litmus.Required
+  alias Litmus.Type
+
+  defstruct required: false
+
+  @type t :: %__MODULE__{
+          required: boolean
+        }
+
+  @spec validate_field(t, String.t(), map) :: {:ok, map} | {:error, String.t()}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- convert(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec convert(t, String.t(), map) :: {:ok, map} | {:error, String.t()}
+  defp convert(%__MODULE__{}, field, params) do
+    cond do
+      !Map.has_key?(params, field) ->
```

Comment:
Curious as to when `params` wouldn't have `field`.

Reply:
Given this check in `Required.validate/3` https://github.com/lob/litmus/blob/master/lib/litmus/required.ex#L6.

Reply:
Oh if it's not required lol. I think I answered my own question.

Reply:
Great! Good question :)